rust/operator-binary/src/crd.rs

```diff
@@ -131,6 +136,57 @@ impl AutoTlsCa {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub enum TlsKeyGeneration {
+    Rsa {
+        #[schemars(schema_with = "TlsKeyGeneration::tls_key_length_schema")]
+        length: u32,
+    },
+}
+
+impl TlsKeyGeneration {
+    pub const TLS_RSA_KEY_LENGTH_2048: u32 = 2048;
+    pub const TLS_RSA_KEY_LENGTH_3072: u32 = 3072;
+    pub const TLS_RSA_KEY_LENGTH_4096: u32 = 4096;
+
+    fn default_tls_key_generation() -> Self {
+        Self::Rsa {
+            length: Self::TLS_RSA_KEY_LENGTH_2048,
+        }
+    }
```

Comment:
question: Why not implement `Default` for this type and use `#[serde(default)]` above?

```rust
impl Default for TlsKeyGeneration {
    fn default() -> Self {
        Self::Rsa {
            length: Self::TLS_RSA_KEY_LENGTH_2048,
        }
    }
}
```

Reply:
done https://github.com/stackabletech/secret-operator/pull/506/commits/43edafe89a65d5242ae346e0e0d7d48c1961404e